aten/src/ATen/native/LinearAlgebraUtils.h

```diff
@@ -151,6 +151,21 @@ static inline int64_t matrixStride(const Tensor& batched_matrices) {
   return batched_matrices.size(-1) * batched_matrices.size(-2);
 }
 
+// This function is similar to at::native::toListOfOptionalTensors,
+// but converts inputs to at::indexing::TensorIndex and outputs a std::vector.
+// The output could then be supplied with at::indexing::Ellipsis or
+// at::indexing::Slice, for example, to perform advanced indexing
+// with the `index` method.
+static inline std::vector<at::indexing::TensorIndex>
+toVectorOfTensorIndices(ArrayRef<Tensor> v, c10::optional<int64_t> len = c10::nullopt) {
+  std::vector<at::indexing::TensorIndex> result;
+  result.reserve(len.has_value() ? len.value() : v.size());
```

Comment:
Why the ternary opreator? Shouldn't it always be `v.size()` given the for loop that we have right after this?
Also, to reserve the size simply use the constructor
```c++
  std::vector<at::indexing::TensorIndex> result(v.size());
```

Reply:
Because it has to comply with the toListOfOptionalTensors.

Reply:
Fair enough. Please add a comment noting this.

Also, the point of the constructor still stands.

Reply:
The constructor will not hold because `TensorIndex` does not have a default constructor.